src/main/java/de/bonndan/nivio/model/ComponentDiff.java

```diff
@@ -0,0 +1,61 @@
+package de.bonndan.nivio.model;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+
+public class ComponentDiff {
+
+    /**
+     * Compares changes in two strings.
+     *
+     * @param a       first
+     * @param b       second
+     * @param key     label
+     * @param changes array of changes
+     */
+    public static void compareStrings(String a, String b, String key, List<String> changes) {
+        if (!org.apache.commons.lang3.StringUtils.equals(a, b)) {
+            changes.add(key + " changed to: " + b);
```

Comment:
Any thoughts on thread safety and reads and writes to the changes `List` in nivio? Does it matter and could we run into problems?

Reply:
Maybe. Ad modifying changes is a side effect. Since every compare method returns void, I'll rewrite them to return changes instead.